addons/website/models/website.py

```diff
@@ -1703,6 +1703,10 @@ def _trigram_enumerate_words(self, search_details, search, limit):
             domain = search_detail['base_domain'].copy()
             fields = set(fields).intersection(model._fields)
 
+            if model_name == "product.template":
```

Comment:
Don't add ecommerce related code on website, this should be part of the website_sale module.
You can write hooks here (method doing nothing) which are overridden in website_sale.
Or spread this behavior to all models having the `is_published` field, instead of targeting just `product.template` (and in this case it could be part of the website module).